### PR TITLE
Fix deferreturn and bolt

### PR DESCRIPTION
**Pull request by hpidcock**
## Description of change

- Fix deferreturn panic on go1.14 on arm64 and ppc64 by disabling optimizations
- Fix unaligned ptrs by moving to go.etcd.io/bbolt

## QA steps

- Run unit tests
- Build for ppc64 and arm64 (juju bins will be quite large)
- Deploy a HA controller (+ anything else you can think to test raft)

## Documentation changes

N/A

## Bug reference

https://golang.org/issue/39049
https://github.com/etcd-io/bbolt/pull/201

### CODE REVIEW

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

File: Makefile [hunk 1]
```diff
@@ -56,7 +56,13 @@ ifeq ($(DEBUG_JUJU), 1)
     COMPILE_FLAGS = -gcflags "all=-N -l"
     LINK_FLAGS = -ldflags "-X $(PROJECT)/version.GitCommit=$(GIT_COMMIT) -X $(PROJECT)/version.GitTreeState=$(GIT_TREE_STATE) -X $(PROJECT)/version.build=$(JUJU_BUILD_NUMBER)"
 else
-    COMPILE_FLAGS =
+ifeq ($(shell go env GOARCH | sed -E 's/.*(arm64|ppc64le|ppc64).*/golang/'), golang)
```
Comment (howbazaar) [inline Makefile:59]: Do we need this for arm64? I thought the bug only impacted 32 bit arm.

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**